src/models/history/tree.ts

```diff
@@ -17,27 +20,55 @@ export const Tree = types.model("Tree", {
   }
 }))
 .actions(self => ({
-  updateTreeAfterSharedModelChanges(options?: {sharedModel: SharedModelType}) {
+  updateTreeAfterSharedModelChanges(options?: {sharedModel: SharedModelType,
+      initialSharedModelMap: SharedModelMapSnapshotOutType}) {
     // If there is no sharedModel param, run the update function on all tiles which
     // have shared model references.
     // If there is a sharedModel param, only run the update function on tiles which
     // have this shared model.
 
     // If we don't have a document let the exception happen so we
     // can track this down
-    const document = (self as any).content as DocumentContentModelType ;
+    const document = (self as any).content as DocumentContentModelType;
+
+    // FIXME: if a shared model is unlinked from a tile we need to notify that
+    // tile. However when this happens we no longer have this tile in the list
+    // of tiles. So I think we need to update things so we pass this removed
+    // tile through. Either by constructing a list of tiles or just passing
+    // this one specific tile.
```

Comment:
I think this comment is outdated since we're now passing in `initialSharedModelMap`.